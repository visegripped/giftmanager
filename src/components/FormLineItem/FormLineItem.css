.formLineItem {
  display: flex;
  gap: 8px 32px;
}

/* mobile only */
@media screen and (max-width: 767px) {
  .formLineItem {
    flex-direction: column;
  }
}
/* tablet + */
@media screen and (min-width: 768px) {
  .formLineItem {
    flex-wrap: nowrap;
    justify-content: center;
    align-items: flex-start;
  }
}