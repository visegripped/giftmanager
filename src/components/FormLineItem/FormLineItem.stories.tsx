import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import FormLineItem from './index';

export default {
  title: 'component/FormLineItem',
  component: FormLineItem,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
} as ComponentMeta<typeof FormLineItem>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof FormLineItem> = (args) => <FormLineItem {...args} />;

export const BasicUsage = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
BasicUsage.args = {
  label: 'Some form input',
  children: <textarea></textarea>,
};
