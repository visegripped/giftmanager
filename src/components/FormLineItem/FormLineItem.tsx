import React from 'react';
import './FormLineItem.css';

interface FormLineItemProps {
  label: string;
  cssClasses?: string;
  children: React.ReactNode;
}

const FormLineItem = (props: FormLineItemProps) => {
  const { label, cssClasses, children } = props;
  const className = `formLineItem ${cssClasses}`;
  return (
    <label className={className} data-testid="FormLineItem">
      <span>{label}</span>
      {children}
    </label>
  );
};

export default FormLineItem;
