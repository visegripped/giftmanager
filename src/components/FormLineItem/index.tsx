import FormLineItem from './FormLineItem';

export default FormLineItem;
