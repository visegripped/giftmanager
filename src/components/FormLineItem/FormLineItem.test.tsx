import { render, screen } from '@testing-library/react';
import FormLineItem from './index';

describe('FormLineItem', () => {
  test('snapshot matches', () => {
    render(<FormLineItem cssClasses="test classes" label="this is the input" children="<textarea ></textarea>" />);
    const renderedFormLineItem = screen.getByTestId('FormLineItem');
    expect(renderedFormLineItem).toMatchSnapshot();
  });
});
