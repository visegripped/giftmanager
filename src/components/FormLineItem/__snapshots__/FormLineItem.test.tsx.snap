// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLineItem snapshot matches 1`] = `
<label
  class="formLineItem test classes"
  data-testid="FormLineItem"
>
  <span>
    this is the input
  </span>
  &lt;textarea &gt;&lt;/textarea&gt;
</label>
`;
